kudu/src/main/java/com/yahoo/ycsb/db/KuduYCSBClient.java

```diff
@@ -0,0 +1,322 @@
+  private static final String DEBUG_OPT = "debug";
+  private static final String PRINT_ROW_ERRORS_OPT = "print_row_errors";
+  private static final String PRE_SPLIT_NUM_TABLETS_OPT = "pre_split_num_tablets";
+  private static final String TABLE_NUM_REPLICAS = "table_num_replicas";
+  private static final String BLOCK_SIZE_OPT = "block_size";
+  private static final int BLOCK_SIZE_DEFAULT = 4096;
+  private static final List<String> columnNames = new ArrayList<String>();
+  private static KuduClient client;
+  private static Schema schema;
+  public boolean debug = false;
+  public boolean printErrors = false;
+  public String tableName;
+  private KuduSession session;
+  private KuduTable table;
+
+  @Override
+  public void init() throws DBException {
+    if (getProperties().getProperty(DEBUG_OPT) != null) {
+      this.debug = getProperties().getProperty(DEBUG_OPT).equals("true");
+    }
+    if (getProperties().getProperty(PRINT_ROW_ERRORS_OPT) != null) {
+      this.printErrors = getProperties().getProperty(PRINT_ROW_ERRORS_OPT).equals("true");
+    }
+    if (getProperties().getProperty(PRINT_ROW_ERRORS_OPT) != null) {
+      this.printErrors = getProperties().getProperty(PRINT_ROW_ERRORS_OPT).equals("true");
+    }
+    this.tableName = com.yahoo.ycsb.workloads.CoreWorkload.table;
+    initClient(debug, tableName, getProperties());
+    this.session = client.newSession();
+    if (getProperties().getProperty(SYNC_OPS_OPT) != null &&
+        getProperties().getProperty(SYNC_OPS_OPT).equals("true")) {
+      this.session.setFlushMode(KuduSession.FlushMode.AUTO_FLUSH_SYNC);
+    } else {
+      this.session.setFlushMode(KuduSession.FlushMode.AUTO_FLUSH_BACKGROUND);
+    }
+
+    this.session.setMutationBufferSpace(100);
+    try {
+      this.table = client.openTable(tableName);
+    } catch (Exception e) {
+      throw new DBException("Could not open a table because of:", e);
+    }
+  }
+
+  private synchronized static void initClient(boolean debug, String tableName, Properties prop)
+      throws DBException {
+    if (client != null) return;
+
+    String masterQuorum = prop.getProperty("masterQuorum");
+    if (masterQuorum == null) {
+      masterQuorum = "localhost:7051";
+    }
+
+    int numTablets = getIntFromProp(prop, PRE_SPLIT_NUM_TABLETS_OPT, 4);
+    if (numTablets > MAX_TABLETS) {
+      throw new DBException("Specified number of tablets (" + numTablets + ") must be equal " +
+          "or below " + MAX_TABLETS);
+    }
+
+    int numReplicas = getIntFromProp(prop, TABLE_NUM_REPLICAS, 3);
+
+    int blockSize = getIntFromProp(prop, BLOCK_SIZE_OPT, BLOCK_SIZE_DEFAULT);
+
+    client = new KuduClient.KuduClientBuilder(masterQuorum)
+        .defaultSocketReadTimeoutMs(DEFAULT_SLEEP)
+        .defaultOperationTimeoutMs(DEFAULT_SLEEP)
+        .build();
+    if (debug) {
+      System.out.println("Connecting to the masters at " + masterQuorum);
+    }
+
+    List<ColumnSchema> columns = new ArrayList<ColumnSchema>(11);
```

Comment:
There's no guarantee that there will be 10 non-key columns. Can you make this a variable based on the value of `fieldcount`?